### PR TITLE
Example notebooks in Sphinx documentation

### PR DESCRIPTION
**Pull request by kosack**
This PR makes it possible to add notebooks into the documentation directly using `nbsphinx`.  It moves some of the current examples/notebooks into `docs/examples` and `docs/tutorials`. I also reorganized a bit the documentation tree (to put the API docs at a lower-level)

The features are:

* the notebooks are executed during the doc build (and if fail the tests fail)
* the notebooks now *must* be stored "Stripped" (e.g. with no output), which is nicer for git
* the docs look like this!:

image

This is still a bit of a work in progress, as I need to migrate more of the notebooks to this new scheme (it takes some re-editing), but I think already it's useful.

### CODE REVIEW

File: docs/tutorials/ctapipe_handson.ipynb [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = utils.get_dataset_path(\"gamma_test.simtel.gz\")\n",
```
Comment (dneise) [inline docs/tutorials/ctapipe_handson.ipynb]: I think the return value of `get_dataset_path` is a path-like or string pointing to a file containing *data* .. it is not the *data* itself. Therefore I think this should be named `path`

File: docs/tutorials/ctapipe_handson.ipynb [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+   "outputs": [],
+   "source": [
+    "r0tel.waveform.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "note that this is (N_chan, N_pix, N_samp)"
```
Comment (dneise) [inline docs/tutorials/ctapipe_handson.ipynb]: The output of Box 7 uses this terminology: `n_channels x n_pixels, n_samples`. I think a few more letters might not hurt here. Also I always wondered: **why** does the container help use a an `x` between the first two dimensions and a comma `,` between the last two?

File: docs/tutorials/ctapipe_handson.ipynb [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+    "Now let's write out program"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = utils.get_dataset(\"gamma_test_large.simtel.gz\") \n",
```
Comment (dneise) [inline docs/tutorials/ctapipe_handson.ipynb]: deprecated: --> `get_dataset_path`

File: docs/tutorials/ctapipe_handson.ipynb [hunk 4]
```diff
@@ -0,0 +1,659 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "hillas = pd.read_hdf(\"hillas.h5\", key='/dl1/hillas')\n",
```
Comment (dneise) [inline docs/tutorials/ctapipe_handson.ipynb]: Assumed, somebody just sent me `hillas.h5` via email .. how would I know the key?

Comment (dneise) [reply docs/tutorials/ctapipe_handson.ipynb]: Maybe one could show a bit how dataset exploration works in ctapipe

**Comment by mackaiver** [comment]
Nice work. While I didn't  look at each notebook in detail I think its a good idea 👍

File: docs/tutorials/ctapipe_handson.ipynb (hunk 4 above)
Comment (kosack) [reply docs/tutorials/ctapipe_handson.ipynb]: yes, i think many of these noteboooks will need improvement, for now I'm just moving them over from the old dir and not making large mods other than bug fixes.

To look inside a hdf5 file there are many options `h5ls` probably is the easiest

**Pull request merged**